specs/ClientCertificate.md

```diff
@@ -0,0 +1,380 @@
+                                CHECK_FAILURE(certificate->get_Issuer(&issuer));
+                                wil::unique_cotaskmem_string serialNumber;
+                                CHECK_FAILURE(certificate->get_SerialNumber(&serialNumber));
+                                COREWEBVIEW2_CLIENT_CERTIFICATE_KIND certificateKind;
+                                CHECK_FAILURE(certificate->get_CertificateKind(&certificateKind));
+                                INT64 startDate;
+                                CHECK_FAILURE(certificate->get_ValidStartDate(&startDate));
+                                INT64 expiryDate;
+                                CHECK_FAILURE(certificate->get_ValidExpiryDate(&expiryDate));
+                                ICoreWebView2ClientCertificateIssuerChainList* pemIssuerChainList;
+                                CHECK_FAILURE(certificate->get_PEMEncodedIssuerChainData(&pemIssuerChainList));
+                            }
+                        }
+
+                        if (certificate)
+                        {
+                            CHECK_FAILURE(args->put_SelectedCertificate(certificate));
+                            CHECK_FAILURE(args->put_ClientCertificateRequestResponseState(
+                                COREWEBVIEW2_CLIENT_CERTIFICATE_REQUEST_RESPONSE_STATE_CONTINUE_WITH_CERTIFICATE));
+                        }
+                        else
+                        {
+                            CHECK_FAILURE(args->put_ClientCertificateRequestResponseState(
+                                COREWEBVIEW2_CLIENT_CERTIFICATE_REQUEST_RESPONSE_STATE_CONTINUE_WITHOUT_CERTIFICATE));
+                        }
+                        return S_OK;
+                    })
+                    .Get(),
+                &m_ClientCertificateRequestedToken));
+        }
+        else
+        {
+            CHECK_FAILURE(m_webView->remove_ClientCertificateRequested(
+                m_ClientCertificateRequestedToken));
+        }
+        //! [ClientCertificateRequested]
+    }
+}
+```
+
+## .NET/ WinRT
+```c#
+void ClientCertificateRequestedCmdExecuted(object target, ExecutedRoutedEventArgs e)
+{
+  try
+  {
+    webView.CoreWebView2.ClientCertificateRequested += delegate (
+    object sender, CoreWebView2ClientCertificateRequestedEventArgs args)
+    {
+      List<CoreWebView2ClientCertificate> certificatesList = args.MutuallyTrustedCertificates;
+      if (certificatesList.Count() > 0)
+      {
+        args.SelectedCertificate = certificatesList.FirstOrDefault();
+        args.ClientCertificateRequestResponseState = CoreWebView2ClientCertificateRequestResponseState.ContinueWithCertificate;
+      }
+      else
+      {
+        args.ClientCertificateRequestResponseState = CoreWebView2ClientCertificateRequestResponseState.ContinueWithoutCertificate;
+      }
+    };
+    webView.CoreWebView2.Navigate("uri that requires client certificate for authentication");
```

Comment:
This makes it look like we're setting up the event handler for this navigation. Maybe we can just remove this Navigate call?